cardano-node/src/Cardano/Node/Tracing/Tracers/NodeVersion.hs

```diff
@@ -0,0 +1,137 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+{-# OPTIONS_GHC -Wno-name-shadowing #-}
+
+module Cardano.Node.Tracing.Tracers.NodeVersion
+(
+    NodeVersionTrace (..)
+  , getNodeVersion
+)
+ where
+
+import           Data.Aeson (toJSON, (.=))
+import           Data.Text (Text, pack)
+
+import           Data.Version (Version (..), showVersion)
+import           System.Info (arch, compilerName, compilerVersion, os)
+
+
+import           Cardano.Git.Rev (gitRev)
+import           Cardano.Logging
+
+
+import           Paths_cardano_node (version)
+
+
+
+-- | Node version information
+
+data NodeVersionTrace = NodeVersionTrace
+  { applicationName    :: Text
+  , applicationVersion :: Version
+  , osName             :: Text
+  , architecture       :: Text
+  , compilerName       :: Text
+  , compilerVersion    :: Version
+  , gitRevision        :: Text
+  } deriving (Eq, Show)
+
+-- | Get the node version information
+
+getNodeVersion :: NodeVersionTrace
+getNodeVersion =
+  let applicationName = "cardano-node"
+      applicationVersion = version
+      osName = pack os
+      architecture = pack arch
+      compilerName = pack System.Info.compilerName
+      compilerVersion = System.Info.compilerVersion
+
```

Comment:
`fullCompilerVersion` appears first in `base-4.15-0.0`. To reliably get the patch version number, could we use conditionals like
```
#if MIN_VERSION_base(4,15,0)
[...]
```

For older versions, we could add (e.g. for GHC8.10.7) some sensible fallbacks based on the version bundled with a compiler, and/or make the patch version a `Maybe` type.
```
#elif MIN_VERSION_base(4,14,3) [... -> patch version 7]
#elif MIN_VERSION_base(4,14,1) [... -> patch version 2]
```

Reply:
IMHO the version of `base` is the decisive factor - does it provide `fullCompilerVersion` or not? So making a choice based on that is independent of GHC version.

Reply:
Done and tested